h5py/api_functions.txt

```diff
@@ -80,43 +80,43 @@ hdf5:
 
   # === H5D - Dataset API =====================================================
 
-  hid_t     H5Dcreate2(hid_t loc_id, char *name, hid_t type_id, hid_t space_id, hid_t lcpl_id, hid_t dcpl_id, hid_t dapl_id)
-  hid_t     H5Dcreate_anon(hid_t file_id, hid_t type_id, hid_t space_id, hid_t plist_id, hid_t dapl_id)
+  hid_t     H5Dcreate2(hid_t loc_id, char *name, hid_t type_id, hid_t space_id, hid_t lcpl_id, hid_t dcpl_id, hid_t dapl_id) nogil
+  hid_t     H5Dcreate_anon(hid_t file_id, hid_t type_id, hid_t space_id, hid_t plist_id, hid_t dapl_id) nogil
 
-  hid_t     H5Dopen(hid_t file_id, char *name)
-  hid_t     H5Dopen2(hid_t loc_id, char *name, hid_t dapl_id )
-  herr_t    H5Dclose(hid_t dset_id)
+  hid_t     H5Dopen(hid_t file_id, char *name) nogil
+  hid_t     H5Dopen2(hid_t loc_id, char *name, hid_t dapl_id ) nogil
+  herr_t    H5Dclose(hid_t dset_id) nogil
 
-  hid_t     H5Dget_space(hid_t dset_id)
-  herr_t    H5Dget_space_status(hid_t dset_id, H5D_space_status_t *status)
-  hid_t     H5Dget_type(hid_t dset_id)
+  hid_t     H5Dget_space(hid_t dset_id) nogil
+  herr_t    H5Dget_space_status(hid_t dset_id, H5D_space_status_t *status) nogil
+  hid_t     H5Dget_type(hid_t dset_id) nogil
```

Comment:
I'm not sure whether it makes sense to release the GIL for all of these little operations, which must do a tiny amount of I/O if any (I don't know if they access data already in memory). The [Cython docs](https://cython.readthedocs.io/en/latest/src/userguide/external_C_code.html#acquiring-and-releasing-the-gil) say:

> Note that acquiring the GIL is a blocking thread-synchronising operation, and therefore potentially costly. It might not be worth releasing the GIL for minor calculations. Usually, I/O operations and substantial computations in parallel code will benefit from it.

Maybe it makes sense to only release it for the operations that can do significant amounts of I/O, like H5Dread & H5Dwrite?